pkg/statedb/types.go

```diff
@@ -108,6 +108,20 @@ type Table[Obj ObjectConstraints[Obj]] interface {
 	// Writer when given a write transaction returns a table writer
 	// that can be used to modify the table.
 	Writer(tx WriteTransaction) TableReaderWriter[Obj]
+
+	// Initialized returns a channel that is closed when the table
+	// has been fully initialized. This method can only be used
```

Comment:
Might be good to define what "initialized" means here. I think it means all the initializers have completed? But could also be that data is present/... But how does it interact with the other methods, are you allowed to call e.g. `Reader(tx)` before this? Also, terminology:

```suggestion
	// has been fully initialized. This method MUST NOT be used
	// before starting.
```

Reply:
Another term we could consider is `Synchronized`, but that's also confusing. Initialization is what ever the initializer wants it to be. It's a generic synchronization mechanism whose semantics are table-specific.

This does not interact with the other methods in any way. I'll document that. Or should we completely separate this out into a separate API?

Reply:
I think some of the confusion stems from that I thought a table would be synchronized/ready/initialized as soon as control flow reached code that wants to use it, but I guess that's not guaranteed. I suppose in hindsight understandably so, as the table is too generic to know.
Not sure whether others will make the same mistake, and hence whether it'd need calling out explicitly. Would it be weird for that behaviour to be the default, i.e. `Table[Obj]` would be a synced version, and `TableEarly[Obj]`/`TableRaw[Obj]`/`TablePre[Obj]` would be this?